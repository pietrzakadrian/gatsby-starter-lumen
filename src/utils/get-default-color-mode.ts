const getDefaultColorMode = (): "dark" | "light" => {
  if (typeof window === "undefined") {
    return "dark";
  }

  const mql = window.matchMedia("(prefers-color-scheme: dark)");
  const hasMediaQueryPreference = typeof mql.matches !== "undefined";
  if (hasMediaQueryPreference) {
    return mql.matches ? "dark" : "light";
  }

  return "dark";
};

export default getDefaultColorMode;
