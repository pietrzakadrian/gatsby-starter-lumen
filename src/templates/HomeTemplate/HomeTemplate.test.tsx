import React from "react";

import {testUtils} from "@/utils"

import BlogTemplate from "./HomeTemplate";
import * as mocks from "@/mocks";
import {StaticQuery, useStaticQuery} from "gatsby";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;


describe("BlogTemplate", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  it("renders correctly", () => {


    const tree = testUtils
      .createSnapshotsRenderer(<BlogTemplate />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

