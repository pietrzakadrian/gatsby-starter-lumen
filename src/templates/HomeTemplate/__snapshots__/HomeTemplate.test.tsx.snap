// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogTemplate renders correctly 1`] = `null`;
